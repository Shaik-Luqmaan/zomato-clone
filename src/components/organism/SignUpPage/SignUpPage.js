import React , { useState } from 'react'
import {Grid} from '@material-ui/core'
import LoginField from '../../atoms/LoginField/LoginField'
import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "../../atoms/Typography/Typography";
import CloseIcon from '@material-ui/icons/Close';
import Divider from "@material-ui/core/Divider";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {  faGoogle } from "@fortawesome/free-brands-svg-icons";
import auth from "../../../auth/initAuth";

const useStyles = makeStyles((theme) => ({
    root: {
        backgroundColor: "white",
        width: '400px',
        borderRadius: "10px",
        padding: "20px 20px",
    },
    form: {
        width: "100%", 
        marginTop:"30px",
        marginBottom:"30px"
      },
    popup: {
        position: "fixed",
        zIndex: 1,
        left: 0,
        top: 0,
        paddingTop: "10vh",
        backgroundColor: "rgb(0,0,0,0.8)",
        width: "1311px",
        height: "669px",
    },
    button:{
        margin: "15px 0px 10px 0",
        borderRadius: "5px",
        textTransform: "capitalize",
        padding: "5px",
        fontSize: "18px",
        width: "360px",
    },
    divide:{
        marginTop:"10px"
    },
    icons:{
        padding: "7px",
        color: "rgb(255,0,0,.7)",
      },
    red: {
        color: "rgb(255,0,0,.7)",
        paddingLeft: "10px",
        cursor:"pointer"
      },
  }));

function SignUpPage(props) {
    const classes = useStyles();
    
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");

    const _handleEmailChange = (e) => {
      setEmail(e.target.value);
    };

    const _handlePasswordChange = (e) => {
        setPassword(e.target.value);
    };

    const _handleSubmit = (e, data) => {
        e.preventDefault();
        auth.login(email, password);
    };
    const _handleLoginWithGoogle = (e) => {
        e.preventDefault();
        auth.loginWithGoogle();
    };
    return (
    <div className={classes.popup}>
      <Grid container justify="center">
        <Grid container className={classes.root}>
         <Grid item container alignItems="baseline">
            <Grid item sm>
                 <Typography variant="h4">Signup</Typography>
             </Grid>
             <Grid item>
                <CloseIcon fontSize="medium" style={{curosr:"pointer"}}
                onClick={() => props.setDisplay(false)} />
             </Grid>
       
            <form className={classes.form}>

                <LoginField placeholder={"Email"} required
                    onKeyUp={_handleEmailChange}/>
                <br/><br/>
                <LoginField placeholder={"Password"} 
                    type="password" required
                    onKeyUp={_handlePasswordChange}/>

                <Button
                 type="submit"
                 fullWidth
                 color="secondary"
                 className={classes.button}
                 variant="contained" 
                 onClick={_handleSubmit}
                > Sign In
                </Button>

                <Divider variant="middle" className={classes.divide}/>

                <Button variant="outlined" className={classes.button}  
                    onClick={_handleLoginWithGoogle}>
                <FontAwesomeIcon icon={faGoogle} className={classes.icons}  />
                    Continue with Google
                </Button>

                <Divider variant="middle" className={classes.divide}/>

            </form>

              <Grid item className={classes.divide}>
                <Typography variant="subtitle1">
                    Already have an account? 
                </Typography>
              </Grid>
              
              <Grid item>
              <Typography>
                    <span className={classes.red}>Log In</span>
                  </Typography>
              </Grid>
       
            </Grid>
        </Grid>
      </Grid>
    </div>
  )
}

export default SignUpPage
