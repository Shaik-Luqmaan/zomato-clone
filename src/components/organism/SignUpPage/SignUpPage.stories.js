import React from 'react';
import SignUpPage from './SignUpPage'

export default {
    title: 'organisms/SignUpPage',
    component: SignUpPage,
}

const Template = (args) => <SignUpPage  {...args}/>

export const Primary = Template.bind({});
Primary.args = {
    
}