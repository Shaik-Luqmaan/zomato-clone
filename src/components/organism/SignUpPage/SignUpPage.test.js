import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import SignUpPage from "./SignUpPage";
import renderer from "react-test-renderer";

it("renders snapshot correctly", () => {
  const tree = renderer.create(<SignUpPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders componentwithout crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SignUpPage />, div);
});
