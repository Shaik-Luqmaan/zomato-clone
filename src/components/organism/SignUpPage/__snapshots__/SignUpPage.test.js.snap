// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="makeStyles-popup-3"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-baseline"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-true"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4"
          >
            Signup
          </h4>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            style={
              Object {
                "curosr": "pointer",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <form
          className="makeStyles-form-2"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="email"
            onKeyUp={[Function]}
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="outlined-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-9"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <br />
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="password"
            onKeyUp={[Function]}
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="outlined-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-9"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-4 MuiButton-containedSecondary MuiButton-fullWidth"
            data-testid="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
               Sign In
            </span>
          </button>
          <hr
            className="MuiDivider-root makeStyles-divide-5 MuiDivider-middle"
          />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-4"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-google fa-w-16 makeStyles-icons-6"
                data-icon="google"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 488 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Continue with Google
            </span>
          </button>
          <hr
            className="MuiDivider-root makeStyles-divide-5 MuiDivider-middle"
          />
        </form>
        <div
          className="MuiGrid-root makeStyles-divide-5 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1"
          >
            Already have an account?
          </h6>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            <span
              className="makeStyles-red-7"
            >
              Log In
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
